src/Reflection/DataTransferObjectClass.php

```diff
@@ -66,6 +67,15 @@ public function validate(): void
 
     public function isStrict(): bool
     {
-        return $this->isStrict ??= ! empty($this->reflectionClass->getAttributes(Strict::class));
+        $attribute = null;
+
+        $reflectionClass = $this->reflectionClass;
+        while ($attribute === null && $reflectionClass !== false) {
+            $attribute = $reflectionClass->getAttributes(Strict::class)[0] ?? null;
+
+            $reflectionClass = $reflectionClass->getParentClass();
+        }
+
+        return $this->isStrict ??= $attribute !== null;
```

Comment:
This now isn't caching correctly anymore.

Reply:
Can you elaborate? Tests pass.

Reply:
Previously, the value of `$this->isStrict` was only calculated when it wasn't set (i.e. not initialized). Now it's calculated every time (actually the `??=` is now quite useless, if not harmful, as `$this->isStrict` is never updated with the re-calculated value once it's set).

Reply:
oh, state, right... good catch

Reply:
A solution would be to extract the actual calculation into a private method and call it like `return $this->isStrict ??= $this->determineIsStrict();`

Or to wrap everything into a check if `$this->isStrict` is set and only do the calculation if it is not (and not use the `??=` operator anymore).